test/Utilities/copy.jl

```diff
@@ -157,3 +157,32 @@ end
         @test dest.added_constrained[idxmap[vi].value]
     end
 end
+
+
+MOIU.@model(OrderingConstrainedVariablesModel,                          # Name of model
+            (),                                                         # untyped scalar sets
+            (),                                                         #   typed scalar sets
+            (MOI.Nonnegatives, MOI.Nonpositives),                       # untyped vector sets
+            (),                                                         #   typed vector sets
+            (),                                                         # untyped scalar functions
+            (),                                                         #   typed scalar functions
+            (MOI.VectorOfVariables,),                                   # untyped vector functions
+            (),                                                         #   typed vector functions
+            false
+        )
+        
+MOI.supports_add_constrained_variables(::OrderingConstrainedVariablesModel, ::Type{MOI.Nonnegatives}) = true
+MOI.supports_add_constrained_variables(::OrderingConstrainedVariablesModel, ::Type{MOI.Nonpositives}) = false
+
+@testset "Create variables using supports_add_constrained_variable(s) (#987)" begin
+    src = MOIU.Model{Float64}()
+    a, c1 = MOI.add_constrained_variables(src, MOI.Nonpositives(3))
+    c2 = MOI.add_constraint(src, a, MOI.Nonnegatives(3))
+    dest = OrderingConstrainedVariablesModel{Float64}()
+
+    index_map = MOI.copy_to(dest, src)
+
+
+    @test index_map[c1].value == 2
```

Comment:
Could you also try with a model that supports Nonpositives variables and not Nonnegatives variables. Otherwise, i the cost is not use, there is still 50 % chance that the test pass if we are lucky.

Reply:
I had to edit `model.jl` in order to not have an error thrown.

Reply:
Indeed, this is an issue. For `dest`, it's probably better to use a custom `MOI.ModelLike` then. It can be a simple struct with an `MOIU.Model{Float64}` inside which redirects all calls to it except the `supports_...` calls.

Reply:
I created a custom `ModelLike` and fixed an issue with the types as well.